### PR TITLE
ON 1208 user management

### PR DESCRIPTION
**Pull request by davidgouge**
All these changes were PRd into this branch, so this does not need PRing as such, but will need a tick to merge to master obvs.

### CODE REVIEW

File: src/SFA.DAS.AssessorService.Application.Api.Client/ApplyTokenService.cs [hunk 1]
```diff
@@ -17,20 +17,22 @@ public ApplyTokenService(IWebConfiguration configuration, IHostingEnvironment ho
...
-            var appKey = _configuration.ApplyApiAuthentication.ClientSecret;// 
-            var resourceId = _configuration.ApplyApiAuthentication.ResourceId;// 
-
-            var authority = $"https://login.microsoftonline.com/{tenantId}";
-            var clientCredential = new ClientCredential(clientId, appKey);
-            var context = new AuthenticationContext(authority, true);
-            var result = context.AcquireTokenAsync(resourceId, clientCredential).Result;
-
-            return result.AccessToken;
+            return string.Empty;
```
Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.Application.Api.Client/ApplyTokenService.cs]: Do we not need this token anymore?

File: src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Contacts/WhenRetrieveContactsWithPrivileges.cs [hunk 2]
```diff
@@ -19,48 +19,49 @@ public class WhenRetrieveContactsWithPrivileges
...
-            _result.Should().NotBeNull();
-            _result.Should().HaveCount(2);
-            _result.Should().OnlyHaveUniqueItems();
-            var itemResult = _result.FirstOrDefault();
-            itemResult?.Should().NotBeNull();
-            itemResult?.Contact.Should().NotBeNull();
-            itemResult?.Privileges.Should().NotBeNull();
-            itemResult?.Privileges.Should().HaveCountGreaterThan(0);
-        }
+//        [SetUp]
```
Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Contacts/WhenRetrieveContactsWithPrivileges.cs]: What is happening with this unit test now that is commented out?

File: src/SFA.DAS.AssessorService.Application/Handlers/Login/LoginHandler.cs [hunk 3]
```diff
@@ -121,17 +139,18 @@ public async Task<LoginResponse> Handle(LoginRequest request, CancellationToken
 
         private async Task<bool> UserDoesNotHaveAcceptableRole(Guid contactId)
         {
-            var roles = await _contactQueryRepository.GetRolesFor(contactId);
-            return roles.All(r => r.RoleName != "SuperUser");
-                
-            //TODO: This needs to look up the user by the id and check they are in the appropriate role.
-            //return !roles.Contains(_config.Authentication.Role);
+            return false;
```
Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.Application/Handlers/Login/LoginHandler.cs]: Is this function still required?

File: src/SFA.DAS.AssessorService.Application/Handlers/Login/LoginHandler.cs [hunk 4]
```diff
@@ -121,17 +139,18 @@ public async Task<LoginResponse> Handle(LoginRequest request, CancellationToken
...
+//            //TODO: This needs to look up the user by the id and check they are in the appropriate role.
+//            //return !roles.Contains(_config.Authentication.Role);
         }
 
-        private async Task<string> GetUserStatus(string endPointAssessorOrganisationId, Guid signInId)
-        {
-            return await _contactQueryRepository.GetContactStatus(endPointAssessorOrganisationId, signInId);
-        }
+//        private async Task<string> GetUserStatus(string endPointAssessorOrganisationId, Guid signInId)
+//        {
```
Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.Application/Handlers/Login/LoginHandler.cs]: If not required, can we delete it?

File: src/SFA.DAS.AssessorService.Application.Api.Client/ApplyTokenService.cs (hunk 1 above)
Comment (alandburns) [reply src/SFA.DAS.AssessorService.Application.Api.Client/ApplyTokenService.cs]: Not yet. Apply API does not yet have any Authentication. 
This is a stalled branch,
We may have to do progress this for other reasons, but will become redundant when QnA API is ready.

File: src/SFA.DAS.AssessorService.Data/ContactQueryRepository.cs [hunk 5]
```diff
@@ -75,12 +68,16 @@ public async Task<Contact> GetContact(string userName)
...
-                .Include(organisation => organisation.Contacts)
-                .Where(q => q.Status != OrganisationStatus.Deleted)
-                .SelectMany(q => q.Contacts)
-                .Where(q => q.Email == email)
-                .FirstOrDefaultAsync();
+            var contact = await _assessorDbContext.Contacts
+                .Include(c => c.Organisation)
+                .FirstOrDefaultAsync(c => c.Email.Equals(email, StringComparison.InvariantCultureIgnoreCase) && c.Organisation.Status != OrganisationStatus.Deleted);
+            
+//            var contact = await _assessorDbContext.Organisations
```
Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.Data/ContactQueryRepository.cs]: Can this be removed?

File: src/SFA.DAS.AssessorService.Data/ContactQueryRepository.cs [hunk 6]
```diff
@@ -108,14 +105,14 @@ public async Task<bool> CheckContactExists(string userName)
...
-        public async Task<string> GetContactStatus(string endPointAssessorOrganisationId, Guid signInId)
-        {
-            var contactStatus = await _assessorDbContext.Contacts.Where(x =>
-                    x.EndPointAssessorOrganisationId == endPointAssessorOrganisationId && x.SignInId == signInId)
-                .FirstOrDefaultAsync();
-
-            return contactStatus?.Status;
-        }
+//        public async Task<string> GetContactStatus(string endPointAssessorOrganisationId, Guid signInId)
+//        {
```
Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.Data/ContactQueryRepository.cs]: Can this be removed?

File: src/SFA.DAS.AssessorService.Web/StartupConfiguration/AuthenticationStartup.cs [hunk 7]
```diff
@@ -230,31 +233,31 @@ public static class AuthenticationStartup
...
-                                             GivenNames = user.GivenNames,
-                                             OrganisationId = user.OrganisationId,
-                                             EndPointAssessorOrganisationId = user.EndPointAssessorOrganisationId,
-                                             Status = user.Status
-                                         };
-
-                                         await contactClient.AssociateDefaultRolesAndPrivileges(contact);
-                                     }                                     
-                                 }
+//                                 else if(user.EndPointAssessorOrganisationId != null && user.Status == ContactStatus.Live)
```
Comment (martin-hornsby) [inline src/SFA.DAS.AssessorService.Web/StartupConfiguration/AuthenticationStartup.cs]: Is this not needed anymore?

**Comment by davidgouge** [comment]
@martin-hornsby Comments acted upon. 👍

**Pull request merged**